packages/react-sdk/docusaurus/docs/React/02-tutorials/02-audio-room.mdx

```diff
@@ -38,44 +44,90 @@ npm install @stream-io/video-react-sdk
 yarn add @stream-io/video-react-sdk
 ```
 
-5. Have populated `.env` file in the project root
-
-```shell
-VITE_STREAM_API_KEY=<your_api_key>
-# URL from which the JWT is requested
-VITE_TOKEN_PROVIDER_URL=<your_token_provider_url>
-```
-
 With the basic setup of the project and the Stream SDK ready we can dive into the code.
 
 ### The Audio Rooms app structure
 
 The application has basically 4 types of views:
 
-1. **Login page** - here we select and authenticate a user
+1. **Login page (omitted)** - here we select and authenticate a user
 2. **Rooms overview page** - lists all the rooms based on given filter conditions
 3. **Room detail page** - allows to load, join, leave a room and ask for permissions to speak
 4. **Create room view as a modal** - allows to create a room and redirects to the room detail page
 
 In the following sections we will explain how to achieve the actions described above.
 
 :::note
-Routing in the app is done by using 3rd party library [react-router-dom](https://reactrouter.com/en/main)
+Routing in the demo app is done by using 3rd party library [react-router-dom](https://reactrouter.com/en/main).
 :::
 
-### User Authentication
+### User authentication
 
 The first important step is to initialize the Stream Video client. Therefore, we need an API key and a token provider.
 
+<TokenSnippet sampleApp='audio-rooms' displayStyle="credentials"/>
+
+You can store these values in your project's `.env` file. Make sure, your app retrieves these values from the `.env` file:
+
+```bash
+VITE_STREAM_API_KEY=<api_key>
+VITE_STREAM_USER_TOKEN=<token>
+VITE_STREAM_USER_ID=<user_id>
+VITE_STREAM_CALL_ID=<call_id>
+```
+
+The credentials can then be passed to the `StreamVideoClient` constructor:
+
+```tsx
+import { useEffect, useState } from 'react';
+import {
+  ChildrenOnly,
+  StreamVideo,
+  StreamVideoClient,
+} from '@stream-io/video-react-sdk';
+
+export const VideoClientProvider = ({ children }: ChildrenOnly) => {
+  // retrieve the API key, user and token...
+
+  const [client, setClient] = useState<StreamVideoClient>();
+
+  useEffect(() => {
+    if (!user) return;
+    setClient(
+      new StreamVideoClient({
+        apiKey,
+        token,
+        user: {
+          id: user.id,
+          name: user.name ?? user.id,
+          // more user data ...
+        },
+      }),
+    );
+  }, [token, user]);
+
+  if (!client) return null;
+
+  return <StreamVideo client={client}>{children}</StreamVideo>;
+};
+```
+
+
+### Dynamic token retrieval
```

Comment:
I don't think we need this part, given that we don't use dynamic tokens in the tutorial

Reply:
We use them in the app.

Reply:
I have removed this part